components/RoadmapPage/Stage.tsx

```diff
@@ -49,10 +49,12 @@ const Stage: FC<StageProps> = ({
     isEnabled: isLocked || !stageData.date,
   })
 
+  const shouldBeLocked =
+    new Date(stageData.date).getTime() - new Date().getTime() >= 0 || !stageData.date
+
   useEffect(() => {
-    if (new Date(stageData.date).getTime() - new Date().getTime() > 0 || !stageData.date)
-      setIsLocked(true)
-  }, [stageData])
+    setIsLocked(shouldBeLocked)
+  }, [shouldBeLocked])
```

Comment:
does this need to be stored in `useState`?

why note just directly access the `shouldBeLocked` variable and remove the `useEffect` hook and the `isLocked` state variable? 🚮